src/main/kotlin/no/fdk/dataset/preview/service/PreviewService.kt

```diff
@@ -39,54 +38,79 @@ class PreviewService(
         }
     }
 
-    @Throws(Exception::class)
-    private fun readCSV(resourceUrl: String, maxNumberOfRows: Int): Table {
-        if(LOGGER.isDebugEnabled) {
-            LOGGER.debug("Reading CSV for $resourceUrl")
+    private fun getMaxNumberOfRows(rows: Int?): Int {
```

Comment:
noen av disse if-else greiene tror jeg blir enklere å lese med en when
```suggestion
    private fun getMaxNumberOfRows(rows: Int?): Int =
        when {
            rows == null -> DEFAULT_ROWS
            rows > MAX_ROWS -> MAX_ROWS
            else -> rows
        }
```